### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by bording**
The changes in this PR fall into one of two categories:

1. When we switched to `net8.0`, the bin folders of the "plugin" projects no longer contained all of their dependencies, which meant those files never made it into the `deploy` folder. I've added `CopyLocalLockFileAssemblies` to fix this.
2. There were several places that were using the Windows `%TEMP%` environment variable, which is not valid on non-Windows operating systems. These were primarily for development scenarios, but I've made changes to remove the requirement of specifying the configuration values that needed `%TEMP%` to have a useable value.

I also included some formatting/cleanup and re-organized the Audit test projects to match the layout of the other two instance types.

### CODE REVIEW

**Review by danielmarbach — APPROVED** [review]

File: src/ServiceControl/App.config [hunk 1]
```diff
@@ -3,68 +3,49 @@
...
+    <!--<add key="ServiceControl/TransportType" value="SQLServer" />-->
+
+    <add key="ServiceControl/PersistenceType" value="RavenDB" />
+    <add key="ServiceControl/DataSpaceRemainingThreshold" value="10" />
+  </appSettings>
+  <connectionStrings>
+    <!-- DEVS - Pick a transport connection string to match chosen transport above -->
+
+    <!-- Learning -->
+    <add name="NServiceBus/Transport" connectionString="" />
```
Comment (mauroservienti) [inline src/ServiceControl/App.config:31]:
```suggestion
   <!-- If the LearningTransport connectionString is empty, it'll default to the execution directory when running/debugging from the IDE -->
    <add name="NServiceBus/Transport" connectionString="" />
```

File: src/ServiceControl.Monitoring/App.config [hunk 2]
```diff
@@ -1,43 +1,45 @@
...
+    <!--<add key="Monitoring/TransportType" value="MSMQ" />-->
+    <!--<add key="Monitoring/TransportType" value="NetStandardAzureServiceBus" />-->
+    <!--<add key="Monitoring/TransportType" value="RabbitMQ.QuorumConventionalRouting" />-->
+    <!--<add key="Monitoring/TransportType" value="SQLServer" />-->
+  </appSettings>
+  <connectionStrings>
+    <!-- DEVS - Pick a transport connection string to match chosen transport above -->
+
+    <!-- Learning -->
+    <add name="NServiceBus/Transport" connectionString="" />
```
Comment (mauroservienti) [inline src/ServiceControl.Monitoring/App.config:25]:
```suggestion
      <!-- If the LearningTransport connectionString is empty, it'll default to the execution directory when running/debugging from the IDE -->
    <add name="NServiceBus/Transport" connectionString="" />
```

File: src/ServiceControl.Audit/App.config [hunk 3]
```diff
@@ -3,63 +3,47 @@
...
+
+    <!-- DEVS - Pick a persistence to run Auditing instance on. -->
+    <add key="ServiceControl.Audit/PersistenceType" value="InMemory" />
+    <!--<add key="ServiceControl.Audit/PersistenceType" value="RavenDB" />-->
+  </appSettings>
+  <connectionStrings>
+    <!-- DEVS - Pick a transport connection string to match chosen transport above -->
+
+    <!-- Learning -->
+    <add name="NServiceBus/Transport" connectionString="" />
```
Comment (mauroservienti) [inline src/ServiceControl.Audit/App.config:29]:
```suggestion
   <!-- If the LearningTransport connectionString is empty, it'll default to the execution directory when running/debugging from the IDE -->
    <add name="NServiceBus/Transport" connectionString="" />
```

**Review by mauroservienti — APPROVED** [review]
Minor things

**Pull request merged**